third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -2854,32 +2862,45 @@ let uidCounter = 0;
 
 function inject(renderer) {
   const id = ++uidCounter;
-  window.__RECORD_REPLAY_ANNOTATION_HOOK__("react-devtools-hook", "inject");
+  window.__RECORD_REPLAY_ANNOTATION_HOOK__("react-devtools-hook", JSON.stringify({type: "inject"}));
   window.__REACT_DEVTOOLS_SAVED_RENDERERS__.push(renderer);
   return id;
 }
 
 function onCommitFiberUnmount(rendererID, fiber) {
-  window.__RECORD_REPLAY_ANNOTATION_HOOK__("react-devtools-hook", "commit-fiber-unmount");
+  // Unmounts are always one fiber at a time during the commit phase.
+  // Stash the unmounted fibers here, so we can map them to persistent 
+  // object IDs inside of `onCommitFiberRoot` processing in the routine.
+  const unmountedFibersSet = getUnmountedFibersSetForRenderer(rendererID);
+  unmountedFibersSet.add(fiber);
+
+  window.__RECORD_REPLAY_ANNOTATION_HOOK__("react-devtools-hook", JSON.stringify({type: "commit-fiber-unmount"}));
 }
 
 function onCommitFiberRoot(rendererID, root, priorityLevel) {
```

Comment:
Not strictly related to this PR, but could we also add something like
```
const annotationType = "commit-fiber-root";
``` 
in each scope where we trigger the annotation too? It'd be great if the code running at the annotations didn't have to have all that hacky `new Error().stack` processing that I added to decide which function it is inside of.

Reply:
I think either way we're dropping backward-compat with all this new stuff.